modules/samples/producers/jms/src/main/java/org/wso2/carbon/sample/jmsclient/JMSClient.java

```
@@ -49,66 +51,130 @@ public static void main(String[] args) {
+
+                        List<String> messagesList = JMSClientUtil.readFile(filePath);
+                        try {
+                            if (format.equalsIgnoreCase("csv")) {
+                                log.info("Sending Map messages on '" + queueName + "' queue");
+                                JMSClientUtil.publishMapMessage(producer, session, messagesList);
+
+                            } else {
+                                log.info("Sending  " + format + " messages on '" + queueName + "' queue");
+                                JMSClientUtil.publishTextMessage(producer, session, messagesList);
+                            }
+                        } catch (JMSException e) {
+                            log.error("Can not subscribe." + e.getMessage(), e);
+                        } catch (IOException e) {
+                            log.error("Error when reading the data file." + e.getMessage(), e);
+                        } finally {
+                            producer.close();
+                            session.close();
+                            queueConnection.stop();
+                        }
                     }
-                } catch (JMSException e) {
-                    log.error("Can not subscribe." + e.getMessage(), e);
-                } catch (IOException e){
-                    log.error("Error when reading the data file." + e.getMessage(), e);
-                }finally{
-                    producer.close();
-                    session.close();
-                    topicConnection.stop();
+                }else{
+                    log.error("Error when creating connection factory. Please check necessary jar files");
                 }
+            }else{
+                log.error("Enter queue name or topic name to be published!");
             }
         } catch (Exception e) {
-            log.error("Error when publishing message" + e.getMessage(), e);
+            log.error("Error when publishing");
```

Exception trace is swallowed. Please fix.